### PR TITLE
[Feature] Configure starlet cache in storage dir

### LINKED ISSUE #18832: [StarOS] some calculation about BE data capacity
### Steps to reproduce the behavior (Required)
image

image

DataUsedCapacity | AvailCapacity | TotalCapacity | UsedPct | MaxDiskUsedPct
These value‘s calculation needs to be confirmed.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
UNKNOWN-f763a0591
```

### PR DESCRIPTION
**Pull request by TszKitLo40**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18832

## Problem Summary(Required) ：

Starlet cache dir can't be configured. It will locates in `storage_root_path/starlet_cache`.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by sduzh — DISMISSED** [review]

File: be/src/runtime/exec_env.cpp [hunk 1]
```diff
@@ -323,6 +323,14 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
         _lake_update_manager = new lake::UpdateManager(_lake_location_provider, update_mem_tracker());
         _lake_tablet_manager = new lake::TabletManager(_lake_location_provider, _lake_update_manager,
                                                        config::lake_metadata_cache_limit);
+        if (starrocks::config::starlet_cache_dir.empty()) {
```
Comment (kevincai) [inline be/src/runtime/exec_env.cpp:326]: already in `starrocks` namespace, can save some types.

File: be/src/runtime/exec_env.cpp [hunk 2]
```diff
@@ -323,6 +323,14 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
...
+            for (int i = 0; i < store_paths.size(); ++i) {
+                starrocks::config::starlet_cache_dir += store_paths[i].path + "/starlet_cache";
+                if (i != paths.size() - 1) {
```
Comment (kevincai) [inline be/src/runtime/exec_env.cpp:329]: `paths` ? not defined symbol? Please do run build and test yourself, staros related code is not enabled by CI yet.

File: be/src/runtime/exec_env.cpp [hunk 3]
```diff
@@ -323,6 +323,14 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
         _lake_update_manager = new lake::UpdateManager(_lake_location_provider, update_mem_tracker());
         _lake_tablet_manager = new lake::TabletManager(_lake_location_provider, _lake_update_manager,
                                                        config::lake_metadata_cache_limit);
+        if (starrocks::config::starlet_cache_dir.empty()) {
+            for (int i = 0; i < store_paths.size(); ++i) {
```
Comment (kevincai) [inline be/src/runtime/exec_env.cpp:327]: check `src/gutils/strings/join.h`, ease the join of all elements with ':'

something like:
```
config::starlet_cache_dir = JoinStrings(store_paths, ":");
```

**Review by kevincai — DISMISSED** [review]

File: be/src/exec/pipeline/aggregate/spillable_aggregate_blocking_sink_operator.cpp [hunk 4]
```diff
@@ -21,6 +21,7 @@
 #include "exec/pipeline/query_context.h"
 #include "exec/sorting/sorting.h"
 #include "exec/spill/spiller.h"
+#include "exec/spill/spiller.hpp"
```
[the remark this answers is not in the document]
Comment (TszKitLo40) [reply be/src/exec/pipeline/aggregate/spillable_aggregate_blocking_sink_operator.cpp:24]: > introduce unrelated change?

Without this change, I can't compile in my environment. I have also submitted a PR https://github.com/StarRocks/starrocks/pull/19791

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wyb** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**